conans/client/generators/deploy.py

```diff
@@ -43,7 +43,12 @@ def content(self):
                     dst = os.path.normpath(dst)
                     mkdir(os.path.dirname(dst))
                     if os.path.islink(src):
-                        linkto = os.path.relpath(os.readlink(src), os.path.dirname(src))
+                        link_target = os.readlink(src)
+                        if not os.path.isabs(link_target):
+                            link_target = os.path.join(os.path.dirname(src), link_target)
+                        linkto = os.path.relpath(link_target, os.path.dirname(src))
+                        if os.path.isfile(dst):
```

Comment:
This looks wrong. When dst is link already isfile return False, when dst is broken link exists return False too. We need to do `os.path.exists(dst) or os.path.islink(dst)`

Reply:
or as you said, unconditionally. just os.unlink(dst) anyway.

Reply:
I have added it, together with more tests. Actually it was not necessary, because the ``os.symlink()`` actually works when it is a broken symlink. But I agree it is more evident to explicitly remove it here. Please check again, thanks!

Reply:
@memsharded looks good now